### PR TITLE
Move tests into a separate tests folder

### PR DESCRIPTION
**Pull request by mcflugen**
This pull request moves all unit tests from inside the landlab package into a separate folder outside of the package (e.g. the tests inside `landlab/grid/tests` are now in `tests/grid`). This will clean up our package and reduce the amount of stuff that is distributed with landlab.

Still remaining to do:
* [ ] move tests for the components
* [ ] update docs (?)

### CODE REVIEW

**Comment by kbarnhart** [comment]
@mcflugen 🎉 

Few thoughts.

Docs happening over at #1003 (in progress, but they build!!!) I've done my best to keep all those changes INSIDE of the landlab/docs folder. Let's talk Monday about sequence of merging (as I assume some things will need to change in the docs once #997 is merged. There are also a number of doc warnings that I'd like to fix that occur because of bad formatting in component docstrings. I'm ignoring them for now since I wan't to make those change _after_ #956 is merged. 

I think that component test moving should happen in #956 (or in a subsequent PR after this and that PR are merged) as I've had to modify some tests as part of my changes.

**Comment by mcflugen** [comment]
@kbarnhart For the most part, I'm just moving the tests, not making changes to the test files, so it shouldn't be a problem to merge this before #956. However, I'm happy to wait on this and merge it after you've merged #956. I don't think this needs to be included in #956.

**Comment by kbarnhart** [comment]
@mcflugen I think you are right and this should merge first. Looks good!!

File: Makefile [hunk 1]
```diff
@@ -51,11 +51,11 @@ clean-test: ## remove test and coverage artifacts
 	rm -fr .pytest_cache
 
 lint: ## check style with flake8
-	flake8 landlab --exclude=examples
+	flake8 landlab tests --exclude=examples
```
Comment (kbarnhart) [inline Makefile]: I *think* we can remove the `--exclude=examples` as I removed the examples folder. 

I should have done that on #996.

**Comment by kbarnhart** [comment]
A note, coverage decreases b/c the test files (all at 100% were counted towards the total).

**Pull request merged**